Makefile

```diff
@@ -210,6 +210,18 @@ build ${PROTOC_GEN_SWIFT}:
 	@rm Tests/LinuxMain.swift.new
 	${SWIFT} build
 
+# This will get run by any other rule that tries to use the plugin.  This hacks
+# in a check during the library build to ensure that the protoc on the local
+# system is one we expect. This was inspired by the findings that lead to
+#   https://github.com/apple/swift-protobuf/issues/111
+# This likely will need follow up to eventually do a pattern match to ensure it
+# is 3.1 or higher.
+${PROTOC_GEN_SWIFT}: build
+	@if [ "$(shell ${PROTOC} --version)" != "libprotoc 3.1.0" ]; then \
```

Comment:
grep returns success if nothing was selected, so you can use it with an 'if' in /bin/sh, using constructs like this:
```
     if ${PROTOC} --version | grep pattern ; then ...
```